visualization/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h

```diff
@@ -88,7 +88,7 @@ class PlanningSceneDisplay : public rviz::Display
   void queueRenderSceneGeometry();
 
   // pass the execution of this function call to a separate thread that runs in the background
-  void addBackgroundJob(const boost::function<void()> &job, const std::string &name);
+  void addBackgroundJob(const boost::function<void()> &job, const std::string &name, bool ownThread=false);
```

Comment:
I agree that breaking ABI in this code is well-justified by safety concerns.
You are not using `addBackgroundJob` anywhere though. :-)
Also, it might be a good idea to have a separate function name for that, instead of extending the existing one with orthogonal code. `spawnBackgroundJob`?
Alternatively, we could simply stick with the `boost::thread` one-liner you used in this request.
But then it might still be a good idea to have explicit comments above each of these lines to explain why this should not be `addBackgroundJob`.